earth2mip/networks/__init__.py

```diff
@@ -239,7 +239,13 @@ def _iterate(self, x, normalize=True, time=None):
 
             while True:
                 if self.source:
-                    x = self.source(x, self.time_step)
+                    x = self.source(
```

Comment:
this is a pretty big (and undocumented) expansion of API of the source function which used to be just Tensor -> Tensor. now users need to implement all these options.

should revert.

Reply:
This is what I had in mind:
```suggestion
                    x_with_units = x*self.scale + self.center
                    x += self.source(x_with_units) / self.scale
```

I removed the self.time_step, since the name "source" implies a function returning a source to apply, not a function which applies the source.